dashbuilder/dashbuilder-backend/dashbuilder-services/src/main/java/org/dashbuilder/DashbuilderMigrationService.java

```diff
@@ -0,0 +1,154 @@
+
+    private IOService ioService;
+    private FileSystem oldDatasetsFS;
+    private FileSystem datasetsFS;
+    private FileSystem pluginsFS;
+    private FileSystem perspectivesFS;
+    private FileSystem navigationFS;
+
+    public DashbuilderMigrationService() {
+
+    }
+
+    @Inject
+    public DashbuilderMigrationService(
+            final @Named("ioStrategy") IOService ioService,
+            final @Named("datasetsFS") FileSystem datasetsFS,
+            final @Named("pluginsFS") FileSystem pluginsFS,
+            final @Named("perspectivesFS") FileSystem perspectivesFS,
+            final @Named("navigationFS") FileSystem navigationFS) {
+
+        this.ioService = ioService;
+        this.oldDatasetsFS = ioService.getOrNewFileSystem(URI.create("git://system/datasets"),
+                                                          new HashMap<String, Object>());
+        this.datasetsFS = datasetsFS;
+        this.pluginsFS = pluginsFS;
+        this.perspectivesFS = perspectivesFS;
+        this.navigationFS = navigationFS;
+    }
+
+    @PostConstruct
+    private void init() {
+        migrateDatasets(oldDatasetsFS, datasetsFS);
+        migratePerspectives(pluginsFS, perspectivesFS);
+        migrateNavigation(pluginsFS, navigationFS);
+    }
+
+    public void migrateDatasets(FileSystem sourceFS, FileSystem targetFS) {
+        migrate(sourceFS,
+                targetFS,
+                path -> true);
+    }
+
+    public void migratePerspectives(FileSystem sourceFS, FileSystem targetFS) {
+        migrate(sourceFS,
+                targetFS,
+                path -> path.getFileName().toString().startsWith("perspective_layout"));
+    }
+
+    public void migrateNavigation(FileSystem sourceFS, FileSystem targetFS) {
+        migrate(sourceFS,
+                targetFS,
+                path -> path.getFileName().toString().equals("navtree.json"));
+    }
+
+    private void migrate(FileSystem sourceFS, FileSystem targetFS, Predicate<Path> predicate) {
+        LOGGER.info("Start");
+
+        Path sourceRoot = getRoot(sourceFS);
+        Path targetRoot = getRoot(targetFS);
+
+        if (sourceRoot == null) {
+            LOGGER.warn("End: source does not exists");
+            return;
+        }
+
+        if (targetRoot == null) {
+            LOGGER.warn("End: target does not exists");
+            return;
+        }
+
+        LOGGER.info("Move from " + sourceFS.getName() + " to " + targetFS.getName());
+
+        Files.walkFileTree(sourceRoot, new SimpleFileVisitor<Path>() {
+            @Override
+            public FileVisitResult visitFile(final Path path, final BasicFileAttributes attrs) throws IOException {
+                checkNotNull("file", path);
+                checkNotNull("attrs", attrs);
+
+                Path targetPath = targetRoot.resolve(path.toString());
+
+                if (!predicate.test(path)) {
+                    LOGGER.info("Skip file " + path.toString());
```

Comment:
I would change this to DEBUG level. What do you think?

Reply:
Good idea!
Done